casper/src/test/resources/PoSTest.rho

```diff
@@ -706,13 +706,14 @@ match (
           contract closeBlock(ackCh) = {
             new ackCh0, activeValidatorsCh, setBlockData(`rho:test:block:data:set`), blockDataSet in {
               @PoS!("closeBlock", sysAuthToken, *ackCh0) |
-              @PoS!("getActiveValidators", *activeValidatorsCh) |
-              for (@ack              <- ackCh0;
-                   @activeValidators <- activeValidatorsCh) {
-                // For testing we need one of the active validators to be the block sender
-                setBlockData!("sender", activeValidators.nth(0), *blockDataSet) |
-                for (_ <- blockDataSet) {
-                  ackCh!(ack)
+              for (@ack              <- ackCh0) {
+                @PoS!("getActiveValidators", *activeValidatorsCh) |
+                for (@activeValidators <- activeValidatorsCh){
+                  // For testing we need one of the active validators to be the block sender
+                  setBlockData!("sender", activeValidators.nth(0), *blockDataSet) |
+                  for (_ <- blockDataSet) {
+                    ackCh!(ack)
+                  }
```

Comment:
change the logic here to be sequence here to make sure the setting block sender would be an active validators.
Because in some tests, some validators would be slashed or unbonded which would cause the sender is not activate validator.

Reply:
Is this fix related to main topic of PR?

Reply:
I wonder how it worked so far. It seems by pure luck.

Keys for bonded validators are generated each time so it's strange that we didn't seen some of them slashed or unbonded.
https://github.com/rchain/rchain/blob/bf001ddd520818e6dd28bc59ff4f8d82e07370cc/casper/src/test/scala/coop/rchain/casper/genesis/contracts/PoSSpec.scala#L14-L20

Reply:
The random generated key would always get the same prefix byte which would get a determined order of the list every time. In tests, the codes use `nth` which would get the same group of validators(One group is generated ones and another one group is `"1" * 130`, `"2" * 130` and etc. ) each time for bonding and unbonding things. Changing the order make it break.